src/Corax/Indexing/IndexWriter.cs

```diff
@@ -629,7 +629,25 @@ private bool TryDeleteEntry(Slice termSlice, out long entryId)
             FlushBatch();
             return TryDeleteEntry(termSlice, out entryId);
         }
-        
+
+        public void DeleteByPrefix(string term)
+        {
+            using var __ = Slice.From(_transaction.Allocator, term, ByteStringType.Immutable, out var termSlice);
+
+            var fieldsTree = _fieldsTree.CompactTreeFor(_fieldsMapping.GetByFieldId(Constants.IndexWriter.PrimaryKeyFieldId).FieldName);
```

Comment:
Can you wrap the current method with a check to see if any of the current items (in the context of indexing transaction) need to be removed as well?

eg:
```csharp
if (_indexedEntries.Any(id => SliceComparer.StartWith(id, termSlice)) == false)
{
         //current code
}
else
{
         FlushBatch();
         DeleteByPrefix(term);
}
```

Reply:
done